api/prometheus/v1/remote/remote_api.go

```diff
@@ -117,30 +134,58 @@ type vtProtoEnabled interface {
 	MarshalToSizedBufferVT(dAtA []byte) (int, error)
 }
 
+type gogoProtoEnabled interface {
+	Size() (n int)
+	MarshalToSizedBuffer(dAtA []byte) (n int, err error)
+}
+
+// Sort of a hack to identify v2 requests.
+// Under any marshaling scheme, v2 requests have a `Symbols` field of type []string.
+// So would always have a `GetSymbols()` method which doesn't rely on any other types.
+type v2Request interface {
+	GetSymbols() []string
```

Comment:
This seems a bit hacky, but need it for the contentType detection. I wonder if there is a better way here

Reply:
Interesting. I think this is acceptable for now.